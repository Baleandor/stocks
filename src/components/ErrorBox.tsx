export default function ErrorBox({ errorMessage }: { errorMessage: string }) {
  return (
    <div className="m-auto h-fit w-fit border border-red-500 p-2 text-center">
      <span className="text-4xl text-red-700">{errorMessage}</span>
    </div>
  );
}
