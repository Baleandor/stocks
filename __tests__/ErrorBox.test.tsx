import { render, screen } from "@testing-library/react";
import ErrorBox from "~/components/ErrorBox";

describe("ErrorBox component", () => {
  it("renders the error message", () => {
    const testMessage = "This is an error!";

    render(<ErrorBox errorMessage={testMessage} />); // ARRANGE

    const errorElement = screen.getByText(testMessage); // ACT

    expect(errorElement).toBeInTheDocument(); // ASSERT

    expect(errorElement).toHaveClass("text-4xl text-red-700");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <ErrorBox errorMessage="Snapshot error message" />, // ARRANGE
    );

    expect(asFragment()).toMatchSnapshot(); // ASSERT
  });
});
