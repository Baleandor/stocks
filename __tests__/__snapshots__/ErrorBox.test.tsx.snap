// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBox component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="m-auto h-fit w-fit border border-red-500 p-2 text-center"
  >
    <span
      class="text-4xl text-red-700"
    >
      Snapshot error message
    </span>
  </div>
</DocumentFragment>
`;
